packages/asset-swapper/src/utils/market_operation_utils/sampler.ts

```diff
@@ -128,29 +147,39 @@ const samplerOperations = {
         takerFillAmounts: BigNumber[],
     ): BatchedOperation<DexSample[][]> {
         const subOps = sources.map(source => {
+            let batchedOperation;
             if (source === ERC20BridgeSource.Eth2Dai) {
-                return samplerOperations.getEth2DaiSellQuotes(makerToken, takerToken, takerFillAmounts);
+                batchedOperation = samplerOperations.getEth2DaiSellQuotes(makerToken, takerToken, takerFillAmounts);
             } else if (source === ERC20BridgeSource.Uniswap) {
-                return samplerOperations.getUniswapSellQuotes(makerToken, takerToken, takerFillAmounts);
+                batchedOperation = samplerOperations.getUniswapSellQuotes(makerToken, takerToken, takerFillAmounts);
             } else if (source === ERC20BridgeSource.Kyber) {
-                return samplerOperations.getKyberSellQuotes(makerToken, takerToken, takerFillAmounts);
+                batchedOperation = samplerOperations.getKyberSellQuotes(makerToken, takerToken, takerFillAmounts);
+            } else if (source === ERC20BridgeSource.Curve) {
+                // TODO(dekz) best way to pass this data around, in the world of multiple curves at once?
```

Comment:
Not happy with this at the moment, there can be multiple Curve sources with the same tokens, like right now there is USDC/DAI and USDT/USDC/DAI. So it is not easy to tell the `create_order` which `curveAddress` should be used. 

@dorothy-zbornak any thoughts on the right abstraction here? Pass it through `DexSample`? What will PLP look like as well?